Mathlib/Algebra/Category/ModuleCat/Presheaf.lean

```diff
@@ -416,3 +416,56 @@ lemma restrictionApp_toPresheafOfModules {X Y : Cᵒᵖ} (f : X ⟶ Y) :
     PresheafOfModules.restrictionApp f M.toPresheafOfModules = M.map f := rfl
 
 end BundledCorePresheafOfModules
+
+namespace PresheafOfModules
+
+variable (R)
+
+/-- Auxiliary definition for `unit`. -/
+def unitCore : CorePresheafOfModules R where
+  obj X := R.obj X
+  map {X Y} f := by
+    exact
+      { toFun := (R.map f).toFun
+        map_add' := by simp
+        map_smul' := by simp }
+
+/-- The obvious free presheaf of modules of rank `1`. -/
+abbrev unit : PresheafOfModules R := (unitCore R).toPresheafOfModules
+
+lemma unit_map_one {X Y : Cᵒᵖ} (f : X ⟶ Y) : (unit R).map f (1 : R.obj X) = (1 : R.obj Y) :=
+  (R.map f).map_one
+
+variable {R}
+
+/-- The type of sections of a presheaf of modules. -/
+def sections (M : PresheafOfModules.{v} R) : Type _ := (M.presheaf ⋙ forget _).sections
+
+@[ext]
+lemma sections_ext {M : PresheafOfModules.{v} R} (s t : M.sections)
+    (h : ∀ (X : Cᵒᵖ), s.val X = t.val X) : s = t :=
+  Subtype.ext (by ext; apply h)
+
+/-- The bijection `(unit R ⟶ M) ≃ M.sections` for `M : PresheafOfModules R`. -/
+def unitHomEquiv (M : PresheafOfModules R) :
+    (unit R ⟶ M) ≃ M.sections where
+  toFun f :=
+    { val := fun X => f.app X (1 : R.obj X)
+      property := fun {X Y} p => by
+        dsimp
+        erw [← NatTrans.naturality_apply, unit_map_one]
```

Comment:
```suggestion
        erw [← naturality_apply, unit_map_one]
```
The lemma `PresheafOfModules.naturality_apply` seems even closer to what you want in this situation, my hope is that somehow that lemma can be used to remove the need for `erw` but I can't seem to get it to work

Reply:
Thanks!